packages/Bio/src/demo/utils.ts

```diff
@@ -0,0 +1,95 @@
+import * as grok from 'datagrok-api/grok';
+import * as DG from 'datagrok-api/dg';
+import * as ui from 'datagrok-api/ui';
+
+import {_package, sequenceSpaceTopMenu} from '../package';
+import {reduceDimensinalityWithNormalization} from '@datagrok-libraries/ml/src/sequence-space';
+import {StringMetricsNames} from '@datagrok-libraries/ml/src/typed-metrics';
+import {delay} from '@datagrok-libraries/utils/src/test';
+
+export function step(message: string, action: () => Promise<void>, delayMs: number = 1600): () => Promise<void> {
+  return async function() {
+    grok.shell.info(message);
+    const pi = DG.TaskBarProgressIndicator.create(message);
+    try {
+      await action();
+    } finally {
+      pi.close();
+      await delay(delayMs);
+    }
+  };
+}
+
+enum EMBED_COL_NAMES {
+  X = 'Embed_X',
+  Y = 'Embed_Y'
+}
+
+export async function demoSequenceSpace(
+  view: DG.TableView, df: DG.DataFrame, method: string
+): Promise<DG.ScatterPlotViewer> {
+  let resSpaceViewer: DG.ScatterPlotViewer;
+  if (true) {
```

Comment:
this is not the best approach
what is the condition here?

Reply:
comment to clarify switching between two implementations of sequence space